front/src/applications/operationalStudies/views/Scenario/components/ImportTimetableItem/ImportTimetableItemTrainsList.tsx

```
@@ -58,16 +62,45 @@ const ImportTimetableItemTrainsList = ({
   const { t } = useTranslation('operational-studies', { keyPrefix: 'importTrains' });
 
   const {
-    train_schedules: trainSchedulesJsonData,
-    paced_trains: pacedTrainsJsonData,
+    train_schedules: trainSchedulesFromJsonData,
+    paced_trains: pacedTrainsFromJsonData,
     macro_nodes: macroNodes,
   } = trainsJsonData;
+
+  const subCategories = useSubCategoryContext();
+
+  const checkCategory = (category?: TrainCategory | null): TrainCategory | null => {
+    if (!category) return null;
+
+    if (isMainCategory(category)) {
+      const mainCategory = TrainMainCategoryDict.get(category.main_category);
+      return mainCategory ? { ...category, main_category: mainCategory } : null;
```

Do we really need to check main categories?

We don't check anything else in the train schedule: for instance, `comfort` and `constraint_distribution` are just passed through. This bit of code means that we silently discard the category if it's been removed or otherwise doesn't exist.

What makes the main category special?

---

I added this check because, since I check the categories, I figured it wouldn't cost much to check if it already exists.
I'm not opposed to the idea of not checking.

---

I think it's good to check to allow importing with the old category format

---

The old category format can be converted without this check.